src/pages/studyView/charts/barChart/BarChart.tsx

```diff
@@ -74,7 +84,37 @@ export default class BarChart extends React.Component<IBarChartProps, {}>
 
     constructor(props: IBarChartProps) {
         super(props);
+        this.state = {
+            zoomDomain: null,
+            selectedDomain: null,
+        };
         makeObservable(this);
+        this.handleBrush = this.handleBrush.bind(this);
+    }
+
+    componentDidUpdate(prevProps: IBarChartProps) {
+        if (prevProps.showPreviewChecked && !this.props.showPreviewChecked) {
+            this.setState({ zoomDomain: null });
+        }
+    }
+
+    private handleBrush(domain: any) {
+        if (!domain || !domain.x) {
+            console.error('Brush domain is not defined.');
+            return;
+        }
```

Comment:
lets remove this and just check if `domain` and `domain.x` are defined and if so do the actions below.